### PR TITLE
[GSoC'24] DeckPicker: Implementation of Keyboard Shortcuts

### PR DESCRIPTION
**Pull request by SanjaySargam**
## Purpose / Description
Implement keyboard shortcuts to enhance the user experience by providing quick access to commonly used functionalities within the application. This will improve the overall usability, especially for power users who prefer using the keyboard over the mouse.

## Approach
The shortcuts are implemented by adding event listeners for keypress events and mapping them to the respective functions. The approach ensures that the shortcuts are intuitive and do not conflict with existing shortcuts.

| Action | Shortcut in AnkiDroid | Shortcut in Anki |
|--------|--------|--------|
| Open Statistics | T | T |
| Create Deck | D | **n/a**  |
| Check Database | C | C |
| Create Filtered Deck | F | F |
| Deck deck with confirmation  | Del | **n/a** | 
| Delete deck without confirmation | Shift+Del | **n/a**  |
| Rename Deck | R | **n/a**  |
| Restore backup | Ctrl+B | B |
| Open Settings | P | P |
| Check Media | M | M |
| Export collection | Ctrl+E | Ctrl+E |
| Import | Ctrl+Shift+I | Ctrl+Shift+I |
| Open Manage Note types | Ctrl+Shift+N | Ctrl+Shift+N | 

## How Has This Been Tested?
Hp Chromebook

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 1]
```diff
@@ -1362,11 +1365,89 @@ open class DeckPicker :
                     handleDeckSelection(getColUnsafe.decks.selected(), DeckSelectionType.SKIP_STUDY_OPTIONS)
                 }
             }
+            KeyEvent.KEYCODE_T -> {
+                Timber.i("Open Statistics from keypress")
+                openStatistics()
+            }
+            KeyEvent.KEYCODE_D -> {
+                // Ensure it's not Ctrl + D
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1373]: Why is this important for ctrl+d and not other letters?
If a user press ctrl+a for example, we probably don't want to open add-note. This seems like a bug that would be nice to fix for all letters. 
I assume you tested manually, using ctrl+d and ctrl+a other letters and realized that you entered the switch.
Can you please write this in the commit message.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 2]
```diff
@@ -1362,11 +1365,89 @@ open class DeckPicker :
...
+                    Timber.i("check empty cards from keypress")
+                    handleEmptyCards()
+                }
+            }
             else -> {}
         }
         return super.onKeyUp(keyCode, event)
     }
 
+    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1414]: Please explain why some actions are on key up and other on key down. That is far from intuitive to me

**Review by Arthur-Milchior — COMMENTED** [review]
In my opinion, the first commit could be a PR by itself.
If you want to ensure the reviewer spend very little time on it, you could do very simple changes.
First commit, you create openManageNoteType and use it in the shortcut. Second commit, same thing with another method, etc..
So each commit can be reviewed in a second, checking that you just copy pasted code, called the right function, and used the right letter. (We can always squash this to have a single commit in our codebase)

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 1 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1373]: > Why is this important for ctrl+d and not other letters?

Because `Ctrl+D` is a shortcut to check database

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 2 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1414]: onKeyDown() trigger multiple times if the key is held down. So I plan to move remove onKeyDown() and combine in onKeyUp()

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 3]
```diff
@@ -1362,6 +1371,71 @@ open class DeckPicker :
...
+            }
+            KeyEvent.KEYCODE_R -> {
+                // If the `Ctrl` key is pressed along with the `R` key, then trigger the restore backup dialog.
+                // Otherwise, trigger the rename deck dialog when the `R` key is pressed without `Ctrl`.
+                if (event.isCtrlPressed) {
+                    Timber.i("show restore backup dialog from keypress")
+                    showDatabaseErrorDialog(DatabaseErrorDialogType.DIALOG_CONFIRM_RESTORE_BACKUP)
+                } else {
+                    Timber.i("Rename Deck from keypress")
+                    renameDeckDialog(focusedDeck)
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1405]: When we are on a phone and you have almost no indication of which deck is currently selected, I don't think this shortcut is a good idea.
When you are on a split screen, with the deck taking all of the trailiing side of the screen, then it makes sense.

I should add that, at least on simulator, you can navigate in the list of decks with the keyboard's arrows. But the deck selected by the keyboard is only used if you press "enter" (because it simulates a tap), and is not used otherwise. So I'd even state that this is a very misleading behavior.

I think that, before having any action that occurs on the selected deck on phone, we should ensure that it's very clear which deck is currently selected, and that the user can use the keyboard to navigate to it.

I think right now this feature should belong to deckoptions instead of deckpicker

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 4]
```diff
@@ -1362,6 +1371,71 @@ open class DeckPicker :
...
+                    createDeckDialog()
+                }
+            }
+            KeyEvent.KEYCODE_F -> {
+                Timber.i("Create Filtered Deck from keypress")
+                createFilteredDialog()
+            }
+            KeyEvent.KEYCODE_DEL -> {
+                Timber.i("Delete Deck from keypress")
+                deleteDeck(focusedDeck)
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1395]: Same remark as with "Rename" below. But furthermore, I think we should use DeckPickerConfirmDeleteDeckDialog. I opened #16691 to discuss it. 
Unless there is a consensus that this implementation is okay, I really prefer that we only do the deletion if we first ask for the user to confirm the deletion

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 5]
```diff
@@ -2221,6 +2295,28 @@ open class DeckPicker :
         createDeckDialog.showDialog()
     }
 
+    /**
+     * Displays a dialog for creating a new deck.
```
[the remark this answers is not in the document]
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2299]: Ok. I'm happy that you liked my documentation. From now onwards, I'll document each and everything in upcoming PRs

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 3 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1405]: > I think right now this feature should belong to deckoptions instead of deckpicker

You mean `DeckPickerContextMenu` ?

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 4 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1395]: This is still relevant. Without answer  in my bug, let's be safe and ask for confirmation

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 3 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1405]: should I make this shortcut available only if fragmented?

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 4 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1395]: then here should I DeckPickerConfirmDeleteDeckDialog?

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 3 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1405]: I'd prefer yes, please. We'll change it if somebody has an objection

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 4 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1395]: Yes please

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 3 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1405]: also delete shortcut right?

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1405]: Same with the delete shortcut yes.

Please, do longer sentences. Right now, I thought "delete" was the verb and you asked whether I also want you to delete the shortcut.

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerConfirmDeleteDeckDialog.kt [hunk 6]
```diff
@@ -27,12 +28,18 @@ import com.ichi2.utils.BundleUtils.requireLong
...
+    val deckName get() = requireArguments().getString("deckName")
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
         super.onCreate(savedInstanceState)
         return AlertDialog.Builder(requireActivity())
             .setTitle(R.string.delete_deck_title)
-            .setMessage(requireArguments().getString("dialogMessage"))
+            .setMessage(
+                HtmlCompat.fromHtml(
+                    "Are you sure you want to delete <b>$deckName</b>?",
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerConfirmDeleteDeckDialog.kt:39]: Please don't add strings shown to the user in the codebase.
Instead, it should go in `AnkiDrois/src/main/res/values/03-dialogs.xml` with all other strings shown to the user, so that our translators can translate it.

Don't put the `<b>` in the xml file. The translator should not have to know anything about html bold tag. Instead, you'll do `res.getString(R.string.delete_deck_message,  "<b>$deckName</b>")`

Actually, if @david-allison agree, I think we could reuse the sentences that were deleted in 7a65160e0e23e20080091a30abdd4c05fc610e06

```
    <plurals name="delete_deck_message">
        <item quantity="one">Delete all cards in %1$s? It contains %2$d card.</item>
        <item quantity="other">Delete all cards in %1$s? It contains %2$d cards.</item>
    </plurals>
   <string name="delete_cram_deck_message">Send all cards in %s back to their original decks?</string>
```

So that the user know whether they delete a filtered deck or a regular deck, and in the second case, how many card they'll delete (if you see "It contains 1049 cards" you may start to worry!)

In the case of filtered decks (that used to be called "cram deck") I suggest maybe the sentence
> Delete filtered deck <b>%s</b> and send all cards back to their original decks.

So that it's clear it's also a deletion.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 7]
```diff
@@ -1362,6 +1368,82 @@ open class DeckPicker :
...
+                    Timber.i("Create Deck from keypress")
+                    showTheCreateDeckDialog()
+                }
+            }
+            KeyEvent.KEYCODE_F -> {
+                Timber.i("Create Filtered Deck from keypress")
+                createFilteredDialog()
+            }
+            KeyEvent.KEYCODE_DEL -> {
+                // This shortcut should be helpful on large screens only
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1391]: `DEL` should show a dialog

The dialog should note that `SHIFT` + `DEL` would auto-confirm the deletion

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 8]
```diff
@@ -1362,6 +1368,82 @@ open class DeckPicker :
...
+                if (fragmented) {
+                    Timber.i("Delete Deck from keypress")
+                    DeckPickerConfirmDeleteDeckDialog.newInstance(
+                        getColUnsafe.decks.name(focusedDeck),
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1395]: I'd do:

```patch
Index: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt	(revision 08b24ec1aff35e10b765077a2626d16753de8cf2)
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt	(date 1721122589951)
@@ -1391,10 +1391,7 @@
                 // This shortcut should be helpful on large screens only
                 if (fragmented) {
                     Timber.i("Delete Deck from keypress")
-                    DeckPickerConfirmDeleteDeckDialog.newInstance(
-                        getColUnsafe.decks.name(focusedDeck),
-                        focusedDeck
-                    ).show(supportFragmentManager, "deckDeckDialog")
+                    showDeleteDeckConfirmationDialog()
                 }
             }
             KeyEvent.KEYCODE_R -> {
@@ -1449,6 +1446,15 @@
         return super.onKeyUp(keyCode, event)
     }
 
+    private fun showDeleteDeckConfirmationDialog() = launchCatchingTask {
+        val deckName = withCol { decks.name(focusedDeck) }
+        val confirmDeleteDeckDialog = DeckPickerConfirmDeleteDeckDialog.newInstance(
+            deckName = deckName,
+            deckId = focusedDeck
+        )
+        showDialogFragment(confirmDeleteDeckDialog)
+    }
+
     /**
      * Perform the following tasks:
      * Automatic backup
```

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerConfirmDeleteDeckDialog.kt (hunk 6 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerConfirmDeleteDeckDialog.kt:39]: I agree

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 9]
```diff
@@ -1362,11 +1368,113 @@ open class DeckPicker :
...
+            }
+            KeyEvent.KEYCODE_D -> {
+                if (event.isCtrlPressed) {
+                    // Shortcut: CTRL + D
+                    Timber.i("check database from keypress")
+                    showDatabaseErrorDialog(DatabaseErrorDialogType.DIALOG_CONFIRM_DATABASE_CHECK)
+                } else {
+                    // Shortcut: D
+                    Timber.i("Create Deck from keypress")
+                    showTheCreateDeckDialog()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1383]: `displayCreateDeckDialog()`

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 10]
```diff
@@ -1362,11 +1368,113 @@ open class DeckPicker :
...
+                }
+            }
+            KeyEvent.KEYCODE_F -> {
+                Timber.i("Create Filtered Deck from keypress")
+                createFilteredDialog()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1388]: This feels a bit wordy - maybe see if an LLM has better suggestions

```
                displayCreateFilteredDeckDialog()
```

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 11]
```diff
@@ -1362,11 +1368,113 @@ open class DeckPicker :
...
+            }
+            KeyEvent.KEYCODE_DEL -> {
+                // This action on a deck should only occur when the user see the deck name very clearly,
+                // that is, when it appears in the trailing study option fragment
+                if (fragmented) {
+                    if (event.isShiftPressed) {
+                        // Shortcut: Shift + DEL - Delete deck without confirmation dialog
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1395]: Awesome stuff

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 12]
```diff
@@ -1362,11 +1368,113 @@ open class DeckPicker :
...
+            }
+            KeyEvent.KEYCODE_E -> {
+                if (event.isCtrlPressed) {
+                    // Shortcut: CTRL + E
+                    Timber.i("show export dialog from keypress")
+                    exportCollection()
+                } else {
+                    // Shortcut: E
+                    Timber.i("check empty cards from keypress")
+                    handleEmptyCards()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1436]: It feels too common for a very uncommon use-case

Anki doesn't use this shortcut

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 13]
```diff
@@ -1362,11 +1368,113 @@ open class DeckPicker :
...
             else -> {}
         }
         return super.onKeyUp(keyCode, event)
     }
 
+    /**
+     * Displays a confirmation dialog for deleting deck.
+     */
+    private fun showDeleteDeckConfirmationDialog() = launchCatchingTask {
+        val (deckName, totalCards, isFilteredDeck) = withCol {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1462]: really clean

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 14]
```diff
@@ -2221,6 +2329,28 @@ open class DeckPicker :
...
+     *
+     * This function shows a `CreateDeckDialog` for the current `DeckPicker` context,
+     * with the title specified by the resource ID `R.string.new_deck` and the dialog type set to `DECK`.
+     * The dialog does not have a predefined parent deck, hence `null` is passed for the parent deck parameter.
+     *
+     * When a new deck is created through the dialog, the following actions are performed:
+     * 1. The deck list is updated by calling `updateDeckList()`.
+     * 2. The options menu is invalidated, which triggers a refresh of the menu items to reflect the changes.
+     *
+     * Finally, the dialog is displayed by calling `showDialog()` on the `CreateDeckDialog` instance.
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2343]: This feels over-documented - explain what the method does and let the developer read the implementation details if they're unambiguous

@Arthur-Milchior did you request this level of detail? You may want to improve on my suggestion, as it's a little bare

```suggestion
     * Displays a dialog for creating a new deck.
     *
     * @see CreateDeckDialog
```

File: AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.kt [hunk 15]
```diff
@@ -348,6 +346,22 @@ abstract class NavigationDrawerActivity :
...
+    /**
+     * Opens the Statistics Screen.
+     */
+    protected fun openStatistics() {
+        val intent = com.ichi2.anki.pages.Statistics.getIntent(this)
+        startActivity(intent)
+    }
+
+    /**
+     * Opens the Settings Screen.
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.kt:358]: I felt it was a little ambiguous as we have a few settings screens

```suggestion
     * Opens AnkiDroid's Settings Screen.
     * @see Preferences
```

File: AnkiDroid/src/main/res/values/03-dialogs.xml [hunk 16]
```diff
@@ -22,6 +22,11 @@
     <string name="check_db_warning">This may take a long time</string>
     <string name="contextmenu_deckpicker_delete_deck">Delete deck</string>
     <string name="delete_deck_title">Delete deck?</string>
+    <plurals name="delete_deck_message">
+        <item quantity="one">Delete all cards in %1$s? It contains %2$d card.</item>
+        <item quantity="other">Delete all cards in %1$s? It contains %2$d cards.</item>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/03-dialogs.xml:27]: This feels unusual, normally we don't have full stops at the end of strings

Implementer's choice

**Review by david-allison — CHANGES_REQUESTED** [review]
I would find it useful to have a table of keyboard shortcuts, and the equivalent shortcut in Anki Desktop

Some things. such as "Empty Cards" feel out of place

I'd (mildly) recommend that this doesn't have a shortcut, as it's an uncommon action

This could do with a few more `@see` annotations to help with navigation

Code is really high quality, thank you!

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 12 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1436]: > Anki doesn't use this shortcut

Yes, it does not have a shortcut for this. It would be better if we had a shortcut. In the future, we can have a shortcut in Anki too.

**Comment by SanjaySargam** [comment]
> I would find it useful to have a table of keyboard shortcuts, and the equivalent shortcut in Anki Desktop

 @david-allison Created table in PR description

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 8 above)
[the remark this answers is not in the document]
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt]: Oh. I asked your opinion because you appear as author of those functions. Except that I realize now that you probably simply did the kotlin transition. My bad.

Regarding the toast:
* `confirmDeleteDeckDialog` should not be a toast. It's asking the user to confirm the deletion. There is no reason to show it in the notification if the user close the app.
* if the user select the deletion, then we want to keep the already existing snackbar. In particular, because it offers a way to undo. The good news being that this don't require more work from you.


In particular, it seems to me that the documentation of `showDialogFragment` is outdated. I'll correct it.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 12 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1517]: I'd be curious to know why you thing it would be better. IT is not obvious to me.
Personally, I suspect that there will be more users that type "e" by accident than user that really want to handle empty cards.
I especially don't like this because sometime the issue is that the card template is wrong, and you don't even want to delete the cards.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 14 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt]: @david-allison I asked to reduce the quantity of documentation in https://github.com/ankidroid/Anki-Android/pull/16679#discussion_r1667599963.

However, I should note that, in a way, I did ask for too much documentation. This is because, during the first few weeks, Sanjay wrote very little documentation. Even for complex classes such as AnkiFragment. And since Sanjay was still not writing enough documentation, I told him I want to see too much documentation. That he doesn't hesitate to write a lot, we can always cut later. So, yeah, in a sense, I did ask to write that much documentation, as a first step. And later, I tried to explain why I wanted to keep part of the documentation and why we could remove other parts

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 12 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1517]: I understand your concern about accidentally triggering the deletion of empty cards. I want to clarify that pressing 'E' does not immediately delete the empty cards. Instead, it initiates a process that shows a confirmation dialog to the user. This dialog provides details about the empty cards found and allows the user to decide whether to proceed with deletion or not

**Comment by david-allison** [comment]
@SanjaySargam Please feel free to mark my comments as 'resolved' once you've resolved them

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 12 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1517]: I'd vote to remove this

**Review by david-allison — CHANGES_REQUESTED** [review]
I would request that we use Anki Desktop's shortcuts where possible, for compatibility:

| Action | Shortcut in AnkiDroid | Shortcut in Anki |
|--------|--------|--------|
| Check Database | Ctrl + D | C |
| Restore backup | Ctrl+R | B |

I would like to remove 'empty cards' (up to you & @Arthur-Milchior, non-blocking)

In terms of code quality, this is great, thank you!

**Comment by SanjaySargam** [comment]
> Restore backup	Ctrl+R	B

@david-allison Browser cards and Create backup both have shortcut as 'B' in Anki

**Comment by david-allison** [comment]
Ah, it wasn't on the list

* Agreed. Browse is better as 'B'
* I don't see "Restore backup" as `B` in Anki
  * But in AnkiDroid, if we are doing it, put it under a `Ctrl` option, it's a dangerous and potentially irreversible action

**Comment by SanjaySargam** [comment]
> Ah, it wasn't on the list

Shortcut B is already implemented in AnkiDroid and is bound to `Open CardBrowser`

> I don't see "Restore backup" as B in Anki

I mean
image

**Review by david-allison — APPROVED** [review]
Cheers! 

PR description should be updated

**Comment by Arthur-Milchior** [comment]
> I mean image

This only mean that, once the menu is open "b" will select "create backup". It's not generally a shortcut in the deckpicker.
We can ignored those underlined lines
